### PR TITLE
[OPS-1109] Pegasus cluster

### PR DESCRIPTION
**Pull request by mkaito**
See [README](https://github.com/serokell/pegasus-infra/blob/mkaito/ops1109-init/README.md) for details on most things.

new servers:
* Helvetios (serokell.io staging)
* Enif (serokell.io production)
* Matar (PostgreSQL)

### CODE REVIEW

File: servers/helvetios/website.nix [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    createHome = true;
+    home = "/var/lib/${name}";
+  };
+
+  services.oauth2_proxy.nginx.virtualHosts.${name} = {
+    host = "serokell.io";
+    applicationChecksAuth = production;
+  };
+
+  services.nginx.virtualHosts.www = {
```
Comment (balsoft) [inline servers/helvetios/website.nix:39]: Does this work without `serverName`?

File: terraform/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Terraform files for Aquarius
```
Comment (balsoft) [inline terraform/README.md:1]: :pled:

Comment (mkaito) [reply terraform/README.md:1]: :see_no_evil:

File: flake.lock [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+      },
+      "locked": {
+        "lastModified": 1605529100,
+        "narHash": "sha256-wNqupoLDsCMtKiuWXeXBYb+ad+5hgbhSkNz4ichFeBI=",
+        "type": "git",
+        "url": "file:///home/chris/dev/serokell/serokell.nix"
+      },
+      "original": {
+        "type": "git",
+        "url": "file:///home/chris/dev/serokell/serokell.nix"
```
Comment (balsoft) [inline flake.lock:398]: :roll_eyes:

File: servers/helvetios/website.nix (hunk 1 above)
Comment (mkaito) [reply servers/helvetios/website.nix:39]: Added `default = true;` so the hostname shouldn't matter.

File: flake.lock (hunk 3 above)
Comment (mkaito) [reply flake.lock]: Yeah, fixed

File: flake.nix [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+          inherit system;
+          modules =
+            [ config ./common.nix ];
+          specialArgs.inputs = inputs;
+        };
+
+      deployChecks =
+        mapAttrs (_: lib: lib.deployChecks self.deploy) deploy-rs.lib;
+
+      terraformFor = pkgs: pkgs.terraform.withPlugins (p: with p; [ aws ]);
```
Comment (balsoft) [inline flake.nix:37]: I don't really like this, maybe we should just do `pkgs = nixpkgs.legacyPackages.${system}` and then `terraform = pkgs.terraform.withPlugins ...`

Comment (mkaito) [reply flake.nix:37]: 90% of this file is identical to the one in `aquarius-infra`. We should probalby do some deduplication.

File: .buildkite/pipeline.yml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+steps:
+  - label: Check Nix flake
+    commands:
+      - nix-shell --run 'nix --experimental-features "nix-command flakes" flake check --impure'
```
Comment (notgne2) [inline .buildkite/pipeline.yml:4]: What in this repo makes us have to use `--impure`?

**Review by notgne2 — CHANGES_REQUESTED** [review]
All of these files need to be licensed, and given license headers using `reuse`

File: flake.nix [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+  inputs = {
+    nixpkgs.url = "github:serokell/nixpkgs";
+    serokell-nix.url = "github:serokell/serokell.nix";
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+    deploy-rs.url = "github:serokell/deploy-rs";
+    upload-daemon.url = "github:serokell/upload-daemon";
```
Comment (balsoft) [inline flake.nix:12]: Likely not needed

File: flake.nix [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    tezos-packaging = {
+      url = "github:serokell/tezos-packaging";
+      flake = false;
+    };
```
Comment (balsoft) [inline flake.nix:16]: Likely not needed

**Comment by balsoft** [comment]
Please don't put `common` in `servers`.

Also, I think it would be much cleaner to add a `profiles/website` directory and put `backups.nix` and `website.nix` there, export `nixosModules.website` and `nixosModules.backups` from the flake and then import `inputs.self.nixosModules.{website,backups}` from the servers that run the website.

**Comment by mkaito** [comment]
I've moved common modules out to `lib/modules/`, and am passing the path to that via `specialArgs` instead of the flake. I don't think we should expose internal modules on the flake, and this is how `nixpkgs` does it with `modulesPath`.

File: flake.nix [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+      servers = mapAttrs (path: _: import (./servers + "/${path}"))
+        (filterAttrs (_: t: t == "directory") (readDir ./servers));
+      mkSystem = config:
+        nixosSystem {
+          inherit system;
+          modules =
+            [ config ./common.nix ];
+          specialArgs = {
+            inputs = inputs;
+          } // optionalAttrs (builtins.pathExists ./lib/modules) {
```
Comment (balsoft) [inline flake.nix:28]: Is there a reason to do this here? If we factor this snippet out, we still have to specify `./lib/modules` in `flake.nix` of `-infra` repo

Comment (mkaito) [reply flake.nix:28]: Same problem as with `./servers` and `./common.nix`. Given the lack of lexical closures in nix, we'll have to pass those as arguments, unless you have a better idea.

**Pull request merged**